### DISCUSSION TITLE
Allow inherited abstract indexer of the same signature

### DISCUSSION THREAD

**Discussion by GaryHuan9** [dotnet/csharplang#6206] [Language Ideas] [answered]
Hello!
In the current versions of C# we can declare `abstract` indexers like so:
```cs
public abstract class Base
{
	public abstract float this[int index] { get; }
}
```
Any derived class can create custom `get` behavior by simply inheriting from `Base` and implement the indexer. However, it is currently impossible to extend this behavior with the following:
```cs
public abstract class WritableBase : Base
{
	public abstract float this[int index] { set; }
	// ^ this gives a compiler error!
}
```
Allowing this would give more flexibility to using the indexers and provide greater tools for abstraction. While similar behaviors can be roughly achieve with interfaces, they are often sub-optimal and do not convey a clear sense of inheritance. For example, if we have an interface `IWritableBase`, one using that interface can no longer easily access members inside `Base`. 

Is there any reason why this syntax is currently not allowed?

**Comment by HaloFour** [comment]
You can't override the base member because you are changing its shape.  You can't redeclare (or shadow) the base member because it is abstract and that would make it impossible to provide an implementation.

This is supported with interfaces via explicit implementation.  Explicit overrides are possible in the runtime, but C# doesn't expose it.

**Comment by svick** [comment, marked as answer]
I think this is a subset of https://github.com/dotnet/csharplang/issues/1568.

**Reply by GaryHuan9** [reply]
I think you are correct! Thank you!
